backend/services/implementations/authService.ts

```diff
@@ -163,6 +163,30 @@ class AuthService implements IAuthService {
     }
   }
 
+  async isAuthorizedToChangeRole(
+    accessToken: string,
+    requestedUserId: string,
+  ): Promise<boolean> {
+    try {
+      const decodedIdToken: firebaseAdmin.auth.DecodedIdToken = await firebaseAdmin
+        .auth()
+        .verifyIdToken(accessToken, true);
+      const tokenUserId = await this.userService.getUserIdByAuthId(
+        decodedIdToken.uid,
+      );
+
+      const firebaseUser = await firebaseAdmin
+        .auth()
+        .getUser(decodedIdToken.uid);
+
+      return (
+        firebaseUser.emailVerified && String(tokenUserId) !== requestedUserId
```

Comment:
I don't think checking the `emailVerified` field is necessary here, since it can already be checked by `isAuthorizedByRole`. (If you remove the check here, it makes sense to reverse the order of the checks in the `and` in `backend/graphql/index.ts` so that the check runs first.)